dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -154,12 +156,25 @@ public static void main(String[] arguments)
    * @throws RepositoryException when there is a problem in retrieving jar files
    */
   static List<Path> coordinateToJarPaths(String coordinate) throws RepositoryException {
-    DefaultArtifact artifact = new DefaultArtifact(coordinate);
-    DependencyGraph transitiveDependencies =
-        DependencyGraphBuilder.getTransitiveDependencies(artifact);
-    List<DependencyPath> dependencyPaths = transitiveDependencies.list();
+    DefaultArtifact rootArtifact = new DefaultArtifact(coordinate);
+    // dependencyGraph holds multiple versions for one artifact key (groupId:artifactId)
+    DependencyGraph dependencyGraph =
+        DependencyGraphBuilder.getCompleteDependencies(rootArtifact);
+    List<DependencyPath> dependencyPaths = dependencyGraph.list();
+
+    // When building a class path, we only need the first version found in breadth-first search
+    // for each artifact key
+    Set<String> artifactKeySet = new HashSet<>();
+
     List<Path> jarPaths = dependencyPaths.stream().map(dependencyPath -> {
-      File artifactFile = dependencyPath.getLeaf().getFile();
+      Artifact artifact = dependencyPath.getLeaf();
+      // groupId:artifactId
+      String artifactKey = Artifacts.makeKey(artifact);
+      if (artifactKeySet.contains(artifactKey)) {
```

Comment:
It looks like artifactKeySet is empty here? and not used again? What am I missing?

Reply:
Updated source code comment to clarify the purpose.

Reply:
I may have been confused by the use of lambdas here. Would this be clearer if rewritten as a mthod or a for loop?

Reply:
Updated this to use for loop,